source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRChoiceSetViewDataSource.mm

```diff
@@ -10,14 +10,46 @@
 
 using namespace AdaptiveCards;
 
+NSString *checkedCheckboxReuseID = @"checked-checkbox";
+NSString *uncheckedCheckboxReuseID = @"unchecked-checkbox";
+NSString *checkedRadioButtonReuseID = @"checked-radiobutton";
+NSString *uncheckedRadioButtonReuseID = @"unchecked-radiobutton";
+
 const CGFloat padding = 16.0f;
 const CGFloat accessoryViewWidth = 50.0f;
 
+@implementation ACRChoiceSetCell
+
+- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(nullable NSString *)reuseIdentifier
+{
+    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
+    if(self) {
+        UIImage *iconImage = nil;
+        if([reuseIdentifier isEqualToString:@"checked-checkbox"]){
+            iconImage = [UIImage imageNamed:@"checked-checkbox-24.png" inBundle:[NSBundle bundleWithIdentifier:@"MSFT.AdaptiveCards"] compatibleWithTraitCollection:nil];
+        } else if([reuseIdentifier isEqualToString:@"checked-radiobutton"]){
```

Comment:
Are you sure you want to build your own widgets, instead of using native UI elements?

I mean, if that's what you've decided to do, then so be it. I'm just a little surprised because these custom widgets may not feel as natural as native UI. Also may not scale well to larger screens.

Reply:
iOS doesn't have checkboxes and radio buttons. iOS UI experts suggested to use mac style checkboxes and radio buttons. 
vector graphics could have solved the scaling issue, but it seems computationally expensive compared to using just UIImages with recycling cells.



---
In reply to: [211795188](https://github.com/Microsoft/AdaptiveCards/pull/1799#discussion_r211795188) [](ancestors = 211795188)